inlanecruising_plugin/src/object_avoidance.cpp

```diff
@@ -0,0 +1,189 @@
+
+#include <inlanecruising_plugin/object_avoidance.h>
+
+namespace inlanecruising_plugin
+{
+namespace object_avoidance
+{
+    ObjectAvoidance::ObjectAvoidance(){}
+
+    cav_msgs::TrajectoryPlan ObjectAvoidance::update_traj_for_object(cav_msgs::TrajectoryPlan& original_tp, const carma_wm::WorldModelConstPtr& wm_, double current_speed_) 
+    {
+        
+        cav_msgs::TrajectoryPlan update_tpp_vector;
+
+        geometry_msgs::Twist current_velocity;
+        current_velocity.linear.x = current_speed_;
+
+        // // TODO get roadway object
+        std::vector<cav_msgs::RoadwayObstacle> rwol = wm_->getRoadwayObjects();
+        cav_msgs::RoadwayObstacleList rwol2;
+        rwol2.roadway_obstacles = rwol;
+
+        std::cout << "host_vehicle_size" << host_vehicle_size.x << std::endl;
+
+        std::vector<cav_msgs::RoadwayObstacle> rwol_collision = carma_wm::collision_detection::WorldCollisionDetection(rwol2, original_tp, host_vehicle_size, current_velocity, 10.0);// 10 replace to target_time
```

Comment:
Does it make sense to set target time to be the duration of the trajectory plan?

Reply:
I think the trajectory time (6 s) is not sufficient and we would want to detect a collision and start slowing down earlier. But changed that into a parameter that we can change.